### PR TITLE
Treat `.pyw` files as scripts in `uv run` on Windows

### PR DESCRIPTION
**Pull request by zanieb**
Closes https://github.com/astral-sh/uv/issues/6435

### CODE REVIEW

**Comment by zanieb** [comment]
I haven't tested this. Do we need to use `pythonw` instead of `python` or something?

**Comment by samypr100** [comment]
> I haven't tested this. Do we need to use `pythonw` instead of `python` or something?

Yup

**Comment by zanieb** [comment]
Tragic. Will do that then.

**Comment by zanieb** [comment]
Is that right? it should be spawned in the background?

**Comment by zanieb** [comment]
(Yes, I think it is right)

**Comment by samypr100** [comment]
I just tried your branch, seems to work.

**Comment by zanieb** [comment]
Thank you!

File: crates/uv/src/commands/project/run.rs [hunk 1]
```diff
@@ -805,12 +834,18 @@ impl From<&ExternalCommand> for RunCommand {
         let target_path = PathBuf::from(&target);
         if target.eq_ignore_ascii_case("python") {
             Self::Python(args.to_vec())
-        } else if target_path
-            .extension()
-            .is_some_and(|ext| ext.eq_ignore_ascii_case("py"))
-            && target_path.exists()
+        } else if target_path.extension().is_some_and(|ext| {
+            ext.eq_ignore_ascii_case("py") || (cfg!(windows) && ext.eq_ignore_ascii_case("pyw"))
```
Comment (charliermarsh) [inline crates/uv/src/commands/project/run.rs:838]: Should the `||` here be removed? What's the difference between this and the next branch?

Comment (zanieb) [reply crates/uv/src/commands/project/run.rs:838]: Oh that's wrong, yes it should be removed.

Comment (zanieb) [reply crates/uv/src/commands/project/run.rs]: The next branch will use `pythonw.exe` to execute instead of `python.exe`

**Review by charliermarsh — APPROVED** [review]

**Pull request merged**